### PR TITLE
Applying mask during rebalancing

### PR DESCRIPTION
**Pull request by nicholasturner1**
Addressing Issue #50 

Masks and proper subvolumes are passed to _get_balance_weight fn's. Within those fn's, the values are extracted from the value array by means of the mask before rebalance weight computation.

Basic testing completed with model input data

### CODE REVIEW

File: python/front_end/zsample.py [hunk 1]
```diff
@@ -335,18 +349,21 @@ def _prepare_rebalance_weights(self, taffs):
 
         if self.pars['is_rebalance'] or self.pars['is_patch_rebalance']:
             for k, aff in taffs.iteritems():
+
+                msk = tmsks[k] if tmsks[k].size != 0 else np.zeros((3,0,0,0))
```
Comment (jingpengwu) [inline python/front_end/zsample.py]: @nicholasturner1 `msk` was not used here?

Comment (nicholasturner1) [inline python/front_end/zsample.py]: Good catch! Consolidating this code from a stash went a little awry. Should be fixed now.

**Comment by nicholasturner1** [comment]
The original affinity mask algorithm is too slow for large datasets. Will update this with a commit shortly

**Comment by jingpengwu** [comment]
@nicholasturner1 I think you can solve the conflict and create a new pull request. After merging it, we can test the `development` branch. Then, we'll release a new version.

**Pull request merged**